bakery/authorizer.go

```
@@ -0,0 +1,134 @@
+package bakery
+
+import (
+	"golang.org/x/net/context"
+	"gopkg.in/errgo.v1"
+
+	"gopkg.in/macaroon-bakery.v2-unstable/bakery/checkers"
+)
+
+const Everyone = "everyone"
+
+// Authorizer is used to check whether a given user is allowed
+// to perform a set of operations.
+type Authorizer interface {
+	// Allow checks whether the given identity (which will be nil
+	// when there is no authenticated user) is allowed to perform
+	// the given operations. It should return an error only when
+	// some underlying database operation has failed, not when the
```

I don't think this should necessarily talk about database operations. How about "It should return an error only when the authorization cannot be determined"?

---

Good suggestion, done.